### PR TITLE
Fix bad positioning

### PR DESCRIPTION
**Pull request by sizmailov**
This patch fixes issues with position and scale of object that was scaled manually in inkscape and then re-edited with textext (see #11, #13). 

Now scaling of textext object in inkscape is coupled with "scale factor" in plugin window. This behavior match behavior of native text objects in inkscape, so this is definitely right way to handle it. 

Here are several cycles of `create -> edit in inkscape -> edit content in textext -> scale to 1.0 in textext` with different transformations in inkscape. 

g7354-9

Another improvement is addition of a control which allows to choose new positioning relative to old bounding box. This property is similar to "align" property of native text fields except it has two dimensions.

Here is an example of using this alignment property. The "original text" gets edited, scale and content are changed. Now user can affect the resulting positioning scheme. 

New GTK plugin window:
g7354-10
Positioning after edition: 
g7354-11

I modified only GTK version of plugin window, but TK still works with limited functionality, namely 
- no alignment property, default value ("middle center") always used
- scale step is too big, so edited object may not exactly match in size original

Besides that its fully functional.

Note: this patch includes some cleanup, replacing hand-written procedures with analogues from inkscape.

Also this patch should not break documents which was build with older version of this plugin (I did some basic compatibility tests, but may be more extensive testing is required). 

Please, let me know if there are any issues with this PR.

### CODE REVIEW

**Comment by jcwinkler** [comment]
Thank you very much for this huge bunch of work and the detailed description, :+1: 

I will perform the necessary tests in my environment as soon as possible and will give you feedback.

Regards, Jan

P.S. We have really overseen `simpletransform.py` all the years... Would have saved a lot of work... :weary:

**Comment by leberwurstsaft** [comment]
Regarding `simpletransform.py`, that's something I had noticed a while back but obviously forgotten right away :sleeping:.

Regarding your contribution: Wow, that looks great indeed :+1:

**Comment by jcwinkler** [comment]
I tested the patch under Inkscape 0.92.2, 0.91.1 and 0.48.5 (Windows, Linux) while creating new nodes as well as while editing nodes created in previous versions of TexText. I did not observe any issues related to this patch. Everything works flawlessly and as expected.  

The patch also removes the little jumping of nodes after recompiling them (resulted from a small mistake in the handwritten calculation of the bounding box, I already fixed that on [https://bitbucket.org/jcwinkler/textext/branch/feature/pdf2svg_support](https://bitbucket.org/jcwinkler/textext/branch/feature/pdf2svg_support), but `simpletransform.py` is the cleaner way, of course. 

For the user interface I suggest to give the anchor position box a separate label so the user knows immideately what is meant. Maybe by puttin it into s separate group box. The TkInterface needs proper adaption, too, of course. But this can be done before the final release.

Hence, I suggest to merge this patch into the master.

I attached the 0.8 milestone so we can release this together with the pdf2svg, xetex and luatex stuff which will be most likely be completed next week. 

Is there any special way you would like to be credited, @sizmailov ? Or would you like to see your GitHub pseudonym only? Thank you very much again for your contribution!

**Comment by sizmailov** [comment]
Sure, anchor position should go to a separate section, I used an ad-hoc solution so it's far from perfect.

You could mention me as "Sergei Izmailov"

I have a couple of GUI improvement suggestions. I'm not sure whether it's better to create new issue(s) or discuss it here, so I'll just mention them here:
 1. Don't show or disable "Reset" button for newly created nodes, since it does nothing except generating non-sense error messages when Ok button gets pressed (at least in GTK)
 2. Also "Global" label is misleading, since it actually sets scale to last used. Not sure how to name such action, "Set as previous" sounds right, but it's a bit wordy. Again, don't show button if there is no "previous" node. 
 3. Now error messages from preview show up again when "Ok" button pressed, even if all errors were fixed (at least in GTK). It should be easy to fix.

**Comment by leberwurstsaft** [comment]
„At least in GTK“ sounds like the error messages are missing when using the TK variant. We should investigate that, too.

**Comment by sizmailov** [comment]
I'm sorry, if was not clear about error messages.  

All error messages I'm talking about are either duplicates (p. 3) and should not appear or should not happen (p. 1) at first place.  "At least in GTK" means that I saw this erroneous error messages in GTK, but I did't try Tk. Presumably problems are common for GTK and Tk. 

Maybe I'll manage to fill the issue and write patch to it in the next couple of days.

**Comment by jcwinkler** [comment]
I opened separate issues for the the GUI and error message topics since they are not related with this pull request.

**Comment by jcwinkler** [comment]
Sergei, please wait with any new patches until I have merged the next bunch of code relating pdf2svg. This will introduce major changes and cleanup. Thank you!
